src/test/shell/integration/jvm_flags_escaping_test.sh

```diff
@@ -0,0 +1,307 @@
+        "-Darg19='A'",
+        "-Darg20='\"a\"'",
+        "-Darg21='B C'",
+        "-Darg22='\"b c\"'",
+        "-Darg23='D\"E'",
+        "-Darg24='\"d\"e\"'",
+        "-Darg25='C:\\F G'",
+        "-Darg26='\"C:\\f g\"'",
+        "-Darg27='C:\\H\"I'",
+        "-Darg28='\"C:\\h\"i\"'",
+        "-Darg29='C:\\J\\\"K'",
+        "-Darg30='\"C:\\j\\\"k\"'",
+        "-Darg31='C:\\L M '",
+        "-Darg32='\"C:\\l m \"'",
+        "-Darg33='C:\\N O\\'",
+        "-Darg34='\"C:\\n o\\\"'",
+        "-Darg35='C:\\P Q\\ '",
+        "-Darg36='\"C:\\p q\\ \"'",
+        "-Darg37='C:\\R\\S\\'",
+        "-Darg38='C:\\R x\\S\\'",
+        "-Darg39='\"C:\\r\\s\\\"'",
+        "-Darg40='\"C:\\r x\\s\\\"'",
+        "-Darg41='C:\\T U\\W\\'",
+        "-Darg42='\"C:\\t u\\w\\\"'",
+    ],
+)
+eof
+}
+
+function assert_output_of_the_program_with_many_jvm_flags() {
+  expect_log 'arg0=()'
+  expect_log 'arg1=( )'
+  expect_log 'arg2=(")'
+  expect_log 'arg3=("\\)'
+  expect_log 'arg4=(\\)'
+  expect_log 'arg5=(\\")'
+  expect_log 'arg6=(with space)'
+  expect_log 'arg7=(with^caret)'
+  expect_log 'arg8=(space ^caret)'
+  expect_log 'arg9=(caret^ space)'
+  expect_log 'arg10=(with"quote)'
+  expect_log 'arg11=(with\\backslash)'
+  expect_log 'arg12=(one\\ backslash and \\space)'
+  expect_log 'arg13=(two\\\\backslashes)'
+  expect_log 'arg14=(two\\\\ backslashes \\\\and space)'
+  expect_log 'arg15=(one\\"x)'
+  expect_log 'arg16=(two\\\\"x)'
+  expect_log 'arg17=(a \\ b)'
+  expect_log 'arg18=(a \\" b)'
+  expect_log 'arg19=(A)'
+  expect_log 'arg20=("a")'
+  expect_log 'arg21=(B C)'
+  expect_log 'arg22=("b c")'
+  expect_log 'arg23=(D"E)'
+  expect_log 'arg24=("d"e")'
+  expect_log 'arg25=(C:\\F G)'
+  expect_log 'arg26=("C:\\f g")'
+  expect_log 'arg27=(C:\\H"I)'
+  expect_log 'arg28=("C:\\h"i")'
+  expect_log 'arg29=(C:\\J\\"K)'
+  expect_log 'arg30=("C:\\j\\"k")'
+  expect_log 'arg31=(C:\\L M )'
+  expect_log 'arg32=("C:\\l m ")'
+  expect_log 'arg33=(C:\\N O\\)'
+  expect_log 'arg34=("C:\\n o\\")'
+  expect_log 'arg35=(C:\\P Q\\ )'
+  expect_log 'arg36=("C:\\p q\\ ")'
+  expect_log 'arg37=(C:\\R\\S\\)'
+  expect_log 'arg38=(C:\\R x\\S\\)'
+  expect_log 'arg39=("C:\\r\\s\\")'
+  expect_log 'arg40=("C:\\r x\\s\\")'
+  expect_log 'arg41=(C:\\T U\\W\\)'
+  expect_log 'arg42=("C:\\t u\\w\\")'
+}
+
+function expect_program_runs() {
+  local -r path="$1"
+  (RUNFILES_DIR= \
+   RUNFILES_MANIFEST_FILE= \
+   RUNFILES_MANIFEST_ONLY= \
+   "$path" >&"$TEST_log" ; ) || fail "expected success"
+}
+
+function expect_program_cannot_run() {
+  local -r path="$1"
+  (RUNFILES_DIR= \
+   RUNFILES_MANIFEST_FILE= \
+   RUNFILES_MANIFEST_ONLY= \
+   "$path" >&"$TEST_log" ; ) && fail "expected failure" || true
+}
+
+# ----------------------------------------------------------------------
+# TESTS
+# ----------------------------------------------------------------------
+
+function test_no_windows_jvm_flags_escaping() {
+  local -r pkg="${FUNCNAME[0]}"
```

Comment:
I'd like to keep it like this. This generates a unique package name for every test.

In case the test cleanup logic were flawed (which would certainly be a bug that should be fixed) such that it didn't fully clean up after tests, we'd still run every test with its own package, reducing the chance that tests overwrite each others files.